WasteWar/Assets/Scripts/Camera Controls/Move.cs

```diff
@@ -0,0 +1,40 @@
+﻿using UnityEngine;
+
+public class Move : MonoBehaviour
+{
+    [SerializeField]
+    private Terrain terrain;
+    private Vector3 terrainSize;
+    private Vector3 camPosition;
+    private Transform cam;
+
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        terrainSize = terrain.terrainData.size;
+
+        //setting cam to the center of terrain
+        transform.position = new Vector3(terrainSize.x / 2, CameraConstants.DEFAULT_HEIGHT, terrainSize.z / 2);
+        
+        //variables for readability
+        cam = transform;
+        camPosition = transform.position;
+    }
+
+    void Update()
+    {
+        if (camPosition.z < (terrainSize.z - CameraConstants.WORLD_BORDER)
```

Comment:
each of those ifs should be it's own function, with the name describing the logic behind it.. so if(!borderReached(z)) or whatever